include/rnp/rnp2.h

```diff
@@ -277,10 +282,71 @@ rnp_result_t rnp_key_have_public(rnp_key_handle_t key, bool *result);
 
 /* TODO: function to add a userid to a key */
 
-/* Signature/verification operations */
+/* Signing operations */
 
 /* TODO define functions for password-based encryption */
 
+rnp_result_t rnp_op_sign_create(rnp_op_sign_t *op,
+                                rnp_ffi_t      ffi,
+                                rnp_input_t    input,
+                                rnp_output_t   output);
+
+rnp_result_t rnp_op_sign_cleartext_create(rnp_op_sign_t *op,
+                                          rnp_ffi_t      ffi,
+                                          rnp_input_t    input,
+                                          rnp_output_t   output);
+
+rnp_result_t rnp_op_sign_detached_create(rnp_op_sign_t *op,
+                                         rnp_ffi_t      ffi,
+                                         rnp_input_t    input,
+                                         rnp_output_t   signature);
+
+rnp_result_t rnp_op_sign_add_signer(rnp_op_sign_t op, rnp_key_handle_t key);
+rnp_result_t rnp_op_sign_set_compression(rnp_op_sign_t op, const char *compression, int level);
+rnp_result_t rnp_op_sign_set_armor(rnp_op_sign_t op, bool armored);
+rnp_result_t rnp_op_sign_set_hash_fn(rnp_op_sign_t op, const char *hash);
```

Comment:
Kind of a nitpick but we don't refer to this as `hash_fn` anywhere else in the code base, so I think we should be consistent and use `hash`.

(and if we ever need to disambiguate between a hash function and its output, we can probably use "digest" or "hash output" or something)

Reply:
Agree, thanks. Just copied the notation from previous edits.